flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_Engine.js

```diff
@@ -64,83 +64,32 @@ var A320_Neo_LowerECAM_Engine;
-                }
-            }
+            this.igniterLeft.updateStatus(engineStarting, n2Igniting, n2Percent);
             if (n2Percent > 50) {
                 // Close left valve
                 this.engineLeft.setEngineBleedValveState(false, false);
             }
-            let ignRightTargetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE;
+
             engineStarting = (SimVar.GetSimVarValue("GENERAL ENG STARTER:2", "bool") === 1) ? true : false;
             n2Igniting = (SimVar.GetSimVarValue("TURB ENG IS IGNITING:2", "bool") === 1) ? true : false;
             n2Percent = SimVar.GetSimVarValue("ENG N2 RPM:2", "percent");
-            if (engineStarting && n2Igniting && n2Percent > 18 && n2Percent < 55) {
-                if (this.ignLeftCurrentState == A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A) {
-                    ignRightTargetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.B;
-                } else {
-                    ignRightTargetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A;
-                }
-            }
+            this.igniterRight.updateStatus(engineStarting, n2Igniting, n2Percent);
             if (n2Percent > 50) {
                 // Close right valve
                 this.engineRight.setEngineBleedValveState(false, false);
             }
-            let ignNeedRefreshTitle = false;
-            if (ignLeftTargetState != this.ignLeftCurrentState) {
-                if (ignLeftTargetState != A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE) {
-                    this.ignLeftCurrentState = ignLeftTargetState;
-                }
-                if (this.ignLeftValueText != null) {
-                    this.ignLeftValueText.textContent = this.getIGNStringFromState(ignLeftTargetState);
-                }
-                ignNeedRefreshTitle = true;
-            }
-            if (ignRightTargetState != this.ignRightCurrentState) {
-                if (ignRightTargetState != A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE) {
-                    this.ignRightCurrentState = ignRightTargetState;
-                }
-                if (this.ignRightValueText != null) {
-                    this.ignRightValueText.textContent = this.getIGNStringFromState(ignRightTargetState);
-                }
-                ignNeedRefreshTitle = true;
-            }
-            if (ignNeedRefreshTitle && (this.ignTitleText != null)) {
-                if ((ignLeftTargetState != A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE) || (ignRightTargetState != A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE)) {
+
+            if ((this.igniterLeft.titleUpdateRequired || this.igniterRight.titleUpdateRequired) && (this.ignTitleText != null)) {
+                if ((this.igniterLeft.currentState != A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE) || (this.igniterRight.currentState != A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE)) {
```

Comment:
**thought**: This is a lot of code to decide if some piece of text should be shown or not.
**suggestion**: I would remove the `titleUpdateRequired` bits and add a function to the `EngineIgniter`, such that you can write:

```suggestion
            if (this.igniterLeft.isIgniting() || this.igniterRight.isIgniting()) {
```

The function name makes the intent of this code clearer, and this structure also lets you hide the information within the `EngineIgniter` type.

Reply:
Done. Removed titleUpdateRequired and re-added the igniterOn state. Title is now updated if either igniter is igniting.

Reply:
Maybe I'm misinterpreting things, but can't `_isIgniting` be removed and replaced with the following?

```js
isIgniting() {
    return this.currentState !== Definitions.IGN_STATE.NONE;
}
```

One more side-note: we usually add a blank line in between function declarations.

Reply:
You are interpreting correctly, and I am over complicating, done.